src/Language/PureScript/Errors.hs

```diff
@@ -509,10 +532,20 @@ prettyPrintSingleError full e = prettyPrintErrorMessage <$> onTypesInErrorMessag
     go (PositionedError srcSpan err)   = paras [ line $ "Error at " ++ displaySourceSpan srcSpan ++ ":"
                                                , indent $ go err
                                                ]
+    go (SimpleErrorWrapper sem)        = goSimple sem
 
   line :: String -> Box.Box
   line = Box.text
 
+  suggestions :: ErrorMessage -> [Box.Box]
+  suggestions = suggestions' . unwrapErrorMessage
+    where
+    suggestions' (TypesDoNotUnify t1 (TypeApp t2 _))
+      | [t1, t2] == [tyObject, tyFunction] = [suggestOp "function composition (<<<)"]
```

Comment:
I think these will fix the examples, then let's merge this:

- You also need to test the case where `t1` and `t2` are flipped.
- Matching `tyObject` isn't enough - you also need to look under type applications. Basically, the left-most thing in a nested application should be `tyObject`. Perhaps just write a function `isObject :: Type -> Bool` to test this.
- I think we can solve the more general problem by making the suggestion when either `t1` or `t2` matches `isObject` and the other is any type. Now, this will match too many things, so we should probably change the suggestion. I would just use: "Note that function composition in PureScript is defined using `(<<<)`".